### DISCUSSION TITLE
eslint-plugin: `strict` config

### DISCUSSION THREAD

**Discussion by ipanasenko** [TanStack/query#8347] [Ideas]
I think it would be nice to provide `strinct` and `flat/strict` configs with every rule enabled as `error`.

I'm willing to implement a PR for this, if this idea is accepted

**Comment by TkDodo** [comment]
we have almost all rules set to `error`, except the `no-rest-destructuring` rule because it’s not critical and sometimes doesn’t have a drawback (if `structuralSharing` is turned off). I see no reason to change that.
